MODELS.js

```diff
@@ -67,9 +67,36 @@ const fetchCommentsByArticle = (article_id) => {
     });
 };
 
+const postAndReturnComment = (comment, article_id) => {
+  const { body, username } = comment;
+  // console.log(2);
+  return db
+    .query(
+      `
+     INSERT INTO comments
+     (body, author, article_id)
+     VALUES(
+      $1,$2,$3
+     )
+     RETURNING *;
+    `,
+      [body, username, article_id]
+    )
+    .then(({ rows }) => {
+      if (!rows[0]) {
```

Comment:
You shouldn't need this custom handling as the absences of a body or username will fall foul of the `NOT NULL` constraint in the db. Try removing it and handling the psql errors instead.

Reply:
Wierdly i am actually catching a would be 200 (for an empty array because article_id doesn't exist), so I've left it in for now.